src/ansi-c/parser.y

```diff
@@ -3447,15 +3460,41 @@ postfixing_abstract_declarator:
 parameter_postfixing_abstract_declarator:
           array_abstract_declarator
         | '(' ')'
+          {
+            // Set function name (last declarator) in source location
+            // before parsing function contracts.  Only do this if we
+            // are at a global scope.
+            if (PARSER.current_scope().prefix.empty()) {
```

Comment:
@kroening is there a better way to check if we are at a global scope?

Reply:
I borrowed this methodology from the `add_declarator` function in `ansi_c_parser.cpp`:
```
  // global?
  if(current_scope().prefix.empty())
    ansi_c_declaration.set_is_global(true);
```

I had originally written something like:
`if (PARSER.current_scope() == PARSER.root_scope())`
But there's no equality comparator defined for the scope class.